README.md

```diff
@@ -53,9 +53,15 @@ The [API documentation](https://keystonejs.com/api/) contains a reference for al
 For walk-throughs and discussions, see the [Guides
 documentation](https://www.keystonejs.com/guides/).
 
-## Changelog
+## Version control
 
-You can find changelogs either by browsing our repository, or by using our [interactive changelog explorer](https://changelogs.xyz/@keystonejs/keystone).
+We do our best to follow semver version control within Keystone. This means package versions have 3 numbers. A change in the first number indicates a breaking change, the second number indicates backward compatible feature and the third number indicates a bug fix.
+
+You can find **changelogs** either by browsing our repository, or by using our [interactive changelog explorer](https://changelogs.xyz/@keystonejs/keystone).
+
+A quick note on dependency management: Keystone is organised into a number of small pakages within a monorepo. When packages in the same repository depend on each other, new versions might not be compatible with older versions. If two or more packages are updated, it can result in breaking changes, even though collectively they appear to be non-breaking.
+
+We do our best to catch this, but recommend updating Keystone packages together to avoid any potential conflict. This is especially important to be aware of if you use automated dependency management tools like Greenkeeper.
```

Comment:
```suggestion
We do our best to catch this but recommend updating Keystone packages together to avoid any potential conflict. This is especially important to be aware of if you use automated dependency management tools like Greenkeeper.
```